contracts/optimistic-ethereum/OVM/verification/OVM_FraudVerifier.sol

```diff
@@ -178,13 +179,25 @@ contract OVM_FraudVerifier is iOVM_FraudVerifier, Lib_AddressResolver {
             "Invalid post-state root inclusion proof."
         );
 
-        require(
-            _postStateRoot != transitioner.getPostStateRoot(),
-            "State transition has not been proven fraudulent."
-        );
+        // If the post state root did not match, then there was fraud and we should
+        // delete the batch
+        bool isFraud;
+        if (_postStateRoot != transitioner.getPostStateRoot()) {
```

Comment:
I don't think we need to do this check -- if the transaction wasn't fraudulent then we shouldn't need to punish anyone or reward the users who contributed to the attempted fraud proof. Essentially the same as my comment above within `OVM_BondManager.finalize`.

Reply:
Reverted back to the `require` statement